incubator/validator-avro/src/test/java/io/aklivity/zilla/runtime/validator/avro/AvroValueValidatorTest.java

```diff
@@ -78,55 +80,146 @@ public void shouldVerifyValidAvroEvent()
 
         DirectBuffer data = new UnsafeBuffer();
 
         byte[] bytes = {0x00, 0x00, 0x00, 0x00, 0x79, 0x06, 0x69, 0x64, 0x30, 0x10};
         data.wrap(bytes, 0, bytes.length);
-        assertFalse(validator.read(data, 0, data.capacity()));
+        assertEquals(-1, validator.validate(data, 0, data.capacity(), fragmentFunction));
+    }
+
+    @Test
+    public void shouldReadAvroEventExpectJson()
+    {
+        CatalogConfig catalogConfig = new CatalogConfig("test0", "test", new TestCatalogOptionsConfig(SCHEMA));
+        LongFunction<CatalogHandler> handler = value -> context.attach(catalogConfig);
+        AvroValidatorConfig config = AvroValidatorConfig.builder()
+                .expect("json")
+                .catalog()
+                    .name("test0")
+                        .schema()
+                        .strategy("topic")
+                        .version("latest")
+                        .subject("test-value")
+                        .build()
+                    .build()
+                .build();
+        AvroValueValidator validator = new AvroReadValueValidator(config, resolveId, handler);
+
+        DirectBuffer data = new UnsafeBuffer();
+
+        byte[] bytes = {0x00, 0x00, 0x00, 0x00, 0x09, 0x06, 0x69, 0x64,
+            0x30, 0x10, 0x70, 0x6f, 0x73, 0x69, 0x74, 0x69, 0x76, 0x65};
+        data.wrap(bytes, 0, bytes.length);
+
+        String json = "{" +
+                "\"id\":\"id0\"," +
+                "\"status\":\"positive\"" +
+                "}";
+
+        DirectBuffer expected = new UnsafeBuffer();
+        expected.wrap(json.getBytes(), 0, json.getBytes().length);
+
+        int progress = validator.validate(data, 0, data.capacity(), fragmentFunction);
+        assertEquals(expected.capacity(), progress);
+    }
+
+    @Test
+    public void shouldWriteJsonEventExpectAvro()
+    {
+        CatalogConfig catalogConfig = new CatalogConfig("test0", "test", new TestCatalogOptionsConfig(SCHEMA));
+        LongFunction<CatalogHandler> handler = value -> context.attach(catalogConfig);
+        AvroValidatorConfig config = AvroValidatorConfig.builder()
+                .expect("json")
+                .catalog()
+                    .name("test0")
+                        .schema()
+                        .strategy("topic")
+                        .version("latest")
+                        .subject("test-value")
+                        .build()
+                    .build()
+                .build();
+        AvroValueValidator validator = new AvroWriteValueValidator(config, resolveId, handler);
+
+        DirectBuffer expected = new UnsafeBuffer();
+
+        byte[] bytes = {0x00, 0x00, 0x00, 0x00, 0x01, 0x06, 0x69, 0x64,
+            0x30, 0x10, 0x70, 0x6f, 0x73, 0x69, 0x74, 0x69, 0x76, 0x65};
+        expected.wrap(bytes, 0, bytes.length);
+
+        String payload = "{" +
+                "\"id\":\"id0\"," +
+                "\"status\":\"positive\"" +
+                "}";
```

Comment:
Use indentation aligned with literal JSON layout, including indentation.
See similar feedback with example in JSON validator.